src/renderer/drawer.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from "blockly/core";
+
+export class Drawer extends Blockly.zelos.Drawer {
+  drawStatementInput_(row) {
+    if (this.info_.isBowlerHatBlock()) {
+      // Bowler hat blocks have straight sides with no C-shape/indentation for
+      // statement blocks.
+      this.drawRightSideRow_(row);
+      this.positionStatementInputConnection_(row);
+    } else {
+      super.drawStatementInput_(row);
+    }
+  }
+
+  drawRightSideRow_(row) {
+    if (
+      this.info_.isBowlerHatBlock() &&
+      Blockly.blockRendering.Types.isSpacer(row)
```

Comment:
This looks like it probably runs into problems if your bowlerhat block has more than one row. Can you just comment that explicitly if that's the case?

Reply:
Done!